frontend/test/pytest/test_compiler.py

```diff
@@ -136,6 +138,56 @@ def test_attempts_to_get_inexistent_intermediate_file(self):
         with pytest.raises(ValueError, match="pass .* not found."):
             compiler.get_output_of("inexistent-file")
 
+    def test_runtime_error(self):
+        """Test that an exception is emitted when the runtime raises a C++ exception."""
+
+        class CompileCXXException:
+            """Class that overrides the program to be compiled."""
+
+            _executable = "clang++"
+            _default_flags = ["-shared", "-fPIC", "-x", "c++"]
+
+            @staticmethod
+            def get_output_filename(infile):
+                """Get the name of the output file based on the input file."""
+                return infile.replace(".mlir", ".o")
+
+            # pylint: disable=unused-argument
+            @staticmethod
+            def run(infile, **kwargs):
+                """Run the compilation step."""
+                contents = """
+#include <stdexcept>
+#include <stdarg.h>
```

Comment:
```suggestion
```

Reply:
If you're using some of functions from this lib (I couldn't find any), please use the C++ equivalent of it,
```suggestion
#include <cstdarg>
```